src/display/canvas.js

```diff
@@ -1698,10 +1715,10 @@ var CanvasGraphics = (function CanvasGraphicsClosure() {
         // Only attempt to draw the glyph if it is actually in the embedded font
         // file or if there isn't a font file so the fallback font is shown.
         if (glyph.isInFont || font.missingFile) {
-          if (simpleFillText && !accent) {
+          if (simpleFillText && !accent && this.contentVisible) {
             // common case
             ctx.fillText(character, scaledX, scaledY);
-          } else {
+          } else if (this.contentVisible) {
```

Comment:
It would in my opinion be, a little bit, clearer if you simply enclosed this entire code-block https://github.com/mozilla/pdf.js/blob/6c39aff3743564f3418ba17344e14819f923a427/src/display/canvas.js#L1698-L1717
in `if (this.contentVisible) { ... }`  rather than "inlining" these checks.

Reply:
Also, unless I'm mistaken there's no similar checks in the `showType3Text` below. Don't you need to account for `this.contentVisible` there as well?